### PR TITLE
Form instance loading doesn't handle leaf attributes

### PR DESCRIPTION
**Pull request by phillipm**
Form Instance loading doesn't correctly copy element attributes for leaf nodes.

Fix for http://manage.dimagi.com/default.asp?170392

__Only review/merge after https://github.com/dimagi/javarosa/pull/123 is merged__

### CODE REVIEW

File: core/src/org/javarosa/core/model/instance/TreeElement.java [hunk 1]
```diff
@@ -998,25 +953,27 @@ public void populate(TreeElement incoming, FormDef f) {
...
-            }
+        }
+
+        // copy incoming element's attributes over
+        for (int i = 0; i < incoming.getAttributeCount(); i++) {
+            String name = incoming.getAttributeName(i);
+            String ns = incoming.getAttributeNamespace(i);
+            String value = incoming.getAttributeValue(i);
+
+            this.setAttribute(ns, name, value);
```
Comment (phillipm) [inline core/src/org/javarosa/core/model/instance/TreeElement.java]: The fix: moving this loop out of the else branch of the conditional above, which checks whether a node is a leaf. Attribute copying should happen regardless, right?

**Comment by phillipm** [comment]
@ctsims This is pretty scary! This bug has been around for 3 years; is it really possible that it only came up now?

**Comment by phillipm** [comment]
Also, this fix should probably go into 2.22, but it is hard for me to know the repercussions of this code change, especially since there are no tests around this part of the code. 
I plan to write tests soon, now that I added infrastructure to JavaRosa to load form instances.

**Comment by czue** [comment]
woot!

**Pull request merged**